django/contrib/gis/db/models/functions.py

```diff
@@ -114,23 +117,26 @@ def as_sqlite(self, compiler, connection):
 
 class Area(GeoFunc):
     def as_sql(self, compiler, connection):
-        if connection.ops.oracle:
-            self.output_field = AreaField('sq_m')  # Oracle returns area in units of meters.
-        else:
-            if connection.ops.geography:
-                # Geography fields support area calculation, returns square meters.
-                self.output_field = AreaField('sq_m')
-            elif not self.output_field.geodetic(connection):
-                # Getting the area units of the geographic field.
-                units = self.output_field.units_name(connection)
-                if units:
-                    self.output_field = AreaField(
-                        AreaMeasure.unit_attname(self.output_field.units_name(connection)))
-                else:
-                    self.output_field = FloatField()
+        if connection.ops.geography:
+            # Geography fields support area calculation, returns square meters.
+            self.output_field = AreaField('sq_m')
+        elif not self.output_field.geodetic(connection):
+            # Getting the area units of the geographic field.
+            units = self.output_field.units_name(connection)
+            if units:
+                self.output_field = AreaField(
+                    AreaMeasure.unit_attname(self.output_field.units_name(connection)))
```

Comment:
That code is original Django GIS code, not mine (Only removed few lines without reformatting)